types/types.go

```diff
@@ -292,8 +292,19 @@ const (
 	replicaManagerPrefix  = instanceManagerPrefix + "r-"
 )
 
-func GenerateEngineNameForVolume(vName string) string {
-	return vName + engineSuffix + "-" + util.RandomID()
+func GenerateEngineNameForVolume(vName, currentEngineName string) string {
+	if currentEngineName == "" {
+		return vName + engineSuffix + "-" + "0"
+	}
+	parts := strings.Split(currentEngineName, "-")
+	if len(parts) < 3 {
+		return vName + engineSuffix + "-" + "1"
+	}
```

Comment:
When will we have this case? After the first engine created, isn't the `parts` always 3?

Reply:
I think we should never have the case that `len(parts) < 3`. But since we are using `parts[len(parts)-1]` below, I think we should check to make sure that len(parts) nonempty to avoid panic. Since we need to check for check for len(parts) > 0 and we know that it should be >= 3, I decided to check for `NOT len(parts) < 3`. I can change it back to:
```
	if len(parts) < 1 {
		return vName + engineSuffix + "-" + "1"
	}
```
if preferred. WDYT?

Reply:
Why not just throw an error? because it's not a valid case, it would be better to throw an error?

Since we only expect `vName + engineSuffix + "-" + "1"` if the current volume is `vName + engineSuffix + "-" + "0"`?

Reply:
We also expect `vName + engineSuffix + "-" + "1"` if the current engine is `vName + engineSuffix + "-" + "xxxxxxxxx"` (which is the older format)

Btw, I think you are right that we don't need to check for `len(parts) < 3` or `len(parts) < 1`. The length of `parts` slice is always >= 1 because the `currentEngineName` is != "" at this step. Therefore, I remove the `if len(parts) < 3 {` check. 

WDYT?

Reply:
Sounds good. However, I would like to clarify that checking index out of range is always necessary to prevent any unexpected panic but if it's an invalid case, we should better raise the error explicitly instead.